charts/common/templates/_networkpolicy.yaml

```diff
@@ -0,0 +1,61 @@
+{{- define "common.networkpolicy" -}}
+{{- if .Values.networkpolicy }}
+{{- if .Values.networkpolicy.deploy }}
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ template "library.name" . }}-default
+  labels:
+{{ include "library.labels.standard" . | indent 4 }}
+spec:
+  podSelector: {}
+  policyTypes:
+    - Ingress
+{{- end }}
+{{- end }}
+
+{{- $root := . }}
+{{- range $name, $component := .Values.components }}
+{{- if $component.networkpolicy }}
+{{- $networkpolicy := $component.networkpolicy }}
+{{- if $networkpolicy.deploy }}
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ template "library.name" $root }}-{{ $name }}
+  labels:
+{{ include "library.labels.standard" $root | indent 4 }}
+    app.kubernetes.io/component: {{ $name }}
+spec:
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/name: {{ template "library.name" $root }}
+      app.kubernetes.io/component: {{ $name }}
+  policyTypes:
+    - Ingress
+  ingress:
+    - from:
+      - namespaceSelector:
+          matchLabels:
+            # trusted label is used by Bedag on platformspecific services (e.g. Ingress Controller) for them to access the selected pods
+            trusted: {{ default "false" ($networkpolicy.trusted | quote) }}
```

Comment:
@oliverbaehler do we still need the "trusted" function?

Reply:
@janlauber Just remove that trusted field, to keep it simple.

Reply:
@oliverbaehler I've removed the whole array obj `namespaceSelector`.
Also removed the comment in the `values.yaml` file